### PR TITLE
Convert to multi-module project

### PR DESCRIPTION
**Pull request by aozarov**

### CODE REVIEW

**Comment by jgeewax** [comment]
How does aws-sdk do this? Do they do the same thing? 

What does the install process look like if I want just one piece? What about if I want them all?

**Comment by aozarov** [comment]
Yes, they are following a very similar pattern.
see: https://github.com/aws/aws-sdk-java/

Arie | Ozarov | ozarov@google.com | 415-624-6429

**Comment by aozarov** [comment]
build on Travis and corvalls looks to be understanding the new aggregated format.

File: gcloud-java-core/src/main/java/com/google/gcloud/spi/ServiceRpcProvider.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+package com.google.gcloud.spi;
+
+import com.google.common.collect.Iterables;
+import com.google.gcloud.ServiceOptions;
+
+import java.util.ServiceLoader;
+
+public class ServiceRpcProvider {
+
+  public static <R, O extends ServiceOptions<R, O>> R get(O options,
```
Comment (ludoch) [inline gcloud-java-core/src/main/java/com/google/gcloud/spi/ServiceRpcProvider.java]: Javadoc?

File: gcloud-java-datastore/src/main/java/com/google/gcloud/spi/DatastoreRpcFactory.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.spi;
+
+import com.google.gcloud.datastore.DatastoreServiceOptions;
+
+public interface DatastoreRpcFactory extends
```
Comment (ludoch) [inline gcloud-java-datastore/src/main/java/com/google/gcloud/spi/DatastoreRpcFactory.java]: Javadoc

File: gcloud-java-core/src/main/java/com/google/gcloud/spi/ServiceRpcProvider.java (hunk 1 above)
Comment (aozarov) [inline gcloud-java-core/src/main/java/com/google/gcloud/spi/ServiceRpcProvider.java]: added.

File: gcloud-java-datastore/src/main/java/com/google/gcloud/spi/DatastoreRpcFactory.java (hunk 2 above)
Comment (aozarov) [inline gcloud-java-datastore/src/main/java/com/google/gcloud/spi/DatastoreRpcFactory.java]: Done.

**Comment by aozarov** [comment]
PTAL

File: gcloud-java-core/README.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+Java idiomatic client for [Google Cloud Platform][cloud-platform] services.
+
+[![Build Status](https://travis-ci.org/GoogleCloudPlatform/gcloud-java.svg?branch=master)](https://travis-ci.org/GoogleCloudPlatform/gcloud-java)
+[![Coverage Status](https://coveralls.io/repos/GoogleCloudPlatform/gcloud-java/badge.svg?branch=master)](https://coveralls.io/r/GoogleCloudPlatform/gcloud-java?branch=master)
+
+-  [Homepage] (https://googlecloudplatform.github.io/gcloud-java/)
+-  [API Documentation] (http://googlecloudplatform.github.io/gcloud-java/apidocs)
+-  [Examples] (http://googlecloudplatform.github.io/gcloud-java/apidocs/index.html?com/google/gcloud/examples/package-summary.html)
+
+This moudle provides common functionality and is required by the other service specific modules.
```
Comment (ludoch) [inline gcloud-java-core/README.md]: s/moudle/module/

Comment (aozarov) [inline gcloud-java-core/README.md]: Done.

**Pull request merged**